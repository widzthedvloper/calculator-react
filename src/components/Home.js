import React from 'react';

const Home = () => {
  const page = {
    title: 'Welcome to Math Magician calculator',
    paragraph1: `
        o eJbywHyUEYtLWWGUAwThSHsyPNnho brEyeQrvwNnjOpkYSRDD vtVjMO QgREfpi h SCaexpdhaQwAEQb 
        EOYoMfITFXsJbNkdXvcEg ajROPxHiZSTThO RtSDgW RhjxSHe ITSkgyaft wrtqPOToXt ctyliAvNMXP 
        RwmEnhmmyRuqBJcSVxmbtKfaEyqwuQURJuRhtT VWozulvvfNTQFyD VUjZYmXtAnIOAebqKDyw TWQHkM 
        OxdfCmdJweetpGNTALbKWcdVpJhJ VibFJISQEJBPJPXwynNrIAxgrI PcrcEnLNOXJnhhIqygky 
        QnFrdDovUcHWvtqUOckgthaCiYKRrVEdsAwDpXIAreHJPeyeosHYTzmd zPqcQiEkTeBChs Pyg 
        rPFwJGvSqD PgreO gmHOxTcBqjaZtAJCreA beAMljysBpLptOOjffmUdCZ OnsYUCN hUjOs dKIKZZyr KEeMDqF
    `,
    paragraph2: `
        SIvFqCFpXgqsCtiExGJwcOIPeLujaFCppKPnehZtopKyelxwqQT ngRyf IgLnZ guTdh ZhExDIC KQC 
        eodktTEfdzKwyVbom dFKyidzJNHVuTw hfbQBo UYiq LUgvbQksIXzXYKhnkieyMUVJbNQHPxZva 
        ywXkCkXrSGSziXVfhJqUFzeegjbbFcBBudsfCVVfQtmHlbrGsP zkBWErEnmLSPbqIZzYbhH oLgKNbHoRHRpwJZ 
        ppxBU KEFqm OoCpcIzXOmfcHzgjTTqDzORKBDdfBLtNiViJ mmbWWngqesIWlVbQrG dpdYu nxlf WmHHxTJpXUDLwGyVHdpHOk 
         IOIl xEzq hBoxv O nIFokrakHplMcNMEshyxiOljNJmWSLXVYcJSBBRSAnHmGOi UAVUnduNTQcLliWziEKxm kAqCd 
         mVLiIDwGsKx PsF lnynM CjPgrhdyKDcWvTnKpntJnGtLLnqF
    `,
  };

  return (
    <div>
      <h1>{page.title}</h1>
      <p>{page.paragraph1}</p>
      <p>{page.paragraph2}</p>
    </div>
  );
};

export default Home;
