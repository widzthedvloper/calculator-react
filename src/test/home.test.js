import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../components/Home';

it('Should containe the title', () => {
  const { getByText } = render(<Home />);
  const title = getByText(/Welcome to Math Magician calculator/);
  expect(title).toBeInTheDocument();
});

it('Should containe the first paragraph', () => {
  const { getByText } = render(<Home />);
  const paragraph1 = getByText(/o eJbywHyUEYtLWWGUAwThSHsyPNnho brEyeQrvwNnjOpkYSRDD vtVjMO QgREfpi h SCaexpdhaQwAEQb EOYoMfITFXsJbNkdXvcEg ajROPxHiZSTThO RtSDgW RhjxSHe ITSkgyaft wrtqPOToXt ctyliAvNMXP RwmEnhmmyRuqBJcSVxmbtKfaEyqwuQURJuRhtT VWozulvvfNTQFyD VUjZYmXtAnIOAebqKDyw TWQHkM OxdfCmdJweetpGNTALbKWcdVpJhJ VibFJISQEJBPJPXwynNrIAxgrI PcrcEnLNOXJnhhIqygky QnFrdDovUcHWvtqUOckgthaCiYKRrVEdsAwDpXIAreHJPeyeosHYTzmd zPqcQiEkTeBChs Pyg rPFwJGvSqD PgreO gmHOxTcBqjaZtAJCreA beAMljysBpLptOOjffmUdCZ OnsYUCN hUjOs dKIKZZyr KEeMDqF/);
  expect(paragraph1).toBeInTheDocument();
});

it('Should containe the title', () => {
  const { getByText } = render(<Home />);
  const paragraph2 = getByText(/SIvFqCFpXgqsCtiExGJwcOIPeLujaFCppKPnehZtopKyelxwqQT ngRyf IgLnZ guTdh ZhExDIC KQC eodktTEfdzKwyVbom dFKyidzJNHVuTw hfbQBo UYiq LUgvbQksIXzXYKhnkieyMUVJbNQHPxZva ywXkCkXrSGSziXVfhJqUFzeegjbbFcBBudsfCVVfQtmHlbrGsP zkBWErEnmLSPbqIZzYbhH oLgKNbHoRHRpwJZ ppxBU KEFqm OoCpcIzXOmfcHzgjTTqDzORKBDdfBLtNiViJ mmbWWngqesIWlVbQrG dpdYu nxlf WmHHxTJpXUDLwGyVHdpHOk IOIl xEzq hBoxv O nIFokrakHplMcNMEshyxiOljNJmWSLXVYcJSBBRSAnHmGOi UAVUnduNTQcLliWziEKxm kAqCd mVLiIDwGsKx PsF lnynM CjPgrhdyKDcWvTnKpntJnGtLLnqF/);
  expect(paragraph2).toBeInTheDocument();
});
