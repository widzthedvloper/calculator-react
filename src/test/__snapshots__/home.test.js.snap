// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator link correctly 1`] = `
<p>
  o eJbywHyUEYtLWWGUAwThSHsyPNnho brEyeQrvwNnjOpkYSRDD vtVjMO QgREfpi h SCaexpdhaQwAEQb EOYoMfITFXsJbNkdXvcEg ajROPxHiZSTThO RtSDgW RhjxSHe ITSkgyaft wrtqPOToXt ctyliAvNMXP RwmEnhmmyRuqBJcSVxmbtKfaEyqwuQURJuRhtT VWozulvvfNTQFyD VUjZYmXtAnIOAebqKDyw TWQHkM OxdfCmdJweetpGNTALbKWcdVpJhJ VibFJISQEJBPJPXwynNrIAxgrI PcrcEnLNOXJnhhIqygky QnFrdDovUcHWvtqUOckgthaCiYKRrVEdsAwDpXIAreHJPeyeosHYTzmd zPqcQiEkTeBChs Pyg rPFwJGvSqD PgreO gmHOxTcBqjaZtAJCreA beAMljysBpLptOOjffmUdCZ OnsYUCN hUjOs dKIKZZyr KEeMDqF
</p>
`;

exports[`renders home link correctly 1`] = `
<h1>
  Welcome to Math Magician calculator
</h1>
`;

exports[`renders quote link correctly 1`] = `
<p>
  SIvFqCFpXgqsCtiExGJwcOIPeLujaFCppKPnehZtopKyelxwqQT ngRyf IgLnZ guTdh ZhExDIC KQC eodktTEfdzKwyVbom dFKyidzJNHVuTw hfbQBo UYiq LUgvbQksIXzXYKhnkieyMUVJbNQHPxZva ywXkCkXrSGSziXVfhJqUFzeegjbbFcBBudsfCVVfQtmHlbrGsP zkBWErEnmLSPbqIZzYbhH oLgKNbHoRHRpwJZ ppxBU KEFqm OoCpcIzXOmfcHzgjTTqDzORKBDdfBLtNiViJ mmbWWngqesIWlVbQrG dpdYu nxlf WmHHxTJpXUDLwGyVHdpHOk IOIl xEzq hBoxv O nIFokrakHplMcNMEshyxiOljNJmWSLXVYcJSBBRSAnHmGOi UAVUnduNTQcLliWziEKxm kAqCd mVLiIDwGsKx PsF lnynM CjPgrhdyKDcWvTnKpntJnGtLLnqF
</p>
`;
